src/ape/cli/commands.py

```diff
@@ -0,0 +1,93 @@
+import difflib
+import re
+from typing import Any, Dict
+
+import click
+from click import Context
+
+from ape import networks
+from ape.cli.utils import Abort
+from ape.exceptions import ApeException
+from ape.logging import logger
+from ape.plugins import clean_plugin_name
+
+try:
+    from importlib import metadata  # type: ignore
+except ImportError:
+    import importlib_metadata as metadata  # type: ignore
+
+_DIFFLIB_CUT_OFF = 0.6
+
+
+def network_bound_command(*args, **kwargs):
+    return click.command(cls=NetworkBoundCommand, *args, **kwargs)
+
+
+class NetworkBoundCommand(click.Command):
+    """A command that uses the network option.
+    It will automatically set the network for the duration of the command execution.
+    """
+
+    def invoke(self, ctx: Context) -> Any:
+        with networks.parse_network_choice(ctx.params["network"]):
+            super().invoke(ctx)
+
+
+class ApeCLI(click.MultiCommand):
```

Comment:
I only moved this here because I wanted to get all the custom click stuff in the `cli` folder.
However, it is not really needed and I could put it back.